ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -581,7 +641,14 @@ CURLcode HttpOperation::Setup()
   curl_error_message_[0] = '\0';
   curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_ERRORBUFFER, curl_error_message_);
 
-  rc = SetCurlLongOption(CURLOPT_VERBOSE, 0L);
+  rc = SetCurlLongOption(CURLOPT_VERBOSE, 1L);
```

Comment:
Setting CURLOPT_VERBOSE to 1L is useful for debugging but not sure if it's best practice to keep enabled in production, as it can significantly impact performance in high-throughput scenarios. Would it make sense to make this optional, controlled by an environment variable, configuration option, or macro?

Reply:
Agreed, I think that would make sense as the general rule. 

If the requirement is to expose the TLS version / cipher information in the logs, then `CURLOPT_VERBOSE` is absolutely required in order to use a custom debug function callback, as explained in [CURLOPT_DEBUGFUNCTION](https://curl.se/libcurl/c/CURLOPT_DEBUGFUNCTION.html)

I haven't found anything obvious in terms of option switches / macros that I could repurpose for this so maybe introducing something like a `WITH_CURL_LOGGING` flag would be an acceptable approach.

Reply:
yes `WITH_CURL_LOGGING ` looks good to me.

Reply:
I agree, we need a compile time flag like `WITH_CURL_LOGGING`.

A runtime flag as well, to turn the verbose logging on/off would be nice, but this can be a separate discussion and PR.

Reply:
Apologies for the sudden silence, it's been a busy week so far.

I actually had this similarly implemented with a runtime flag for another project. I think I can have this done in the same PR, just need to run this through the full CI to make sure I didn't miss anything.

If both compile-time and runtime flags are to be supported, the way I see it is that runtime should take precedence as there could be certain cases where this logging info would be useful but, as a general use case, omitting it would fall back to the value set at compile time via `WITH_CURL_LOGGING`.